src/tools/crossgen2/ILCompiler.DependencyAnalysisFramework/DependencyAnalyzer.cs

```diff
@@ -287,7 +287,8 @@ private bool AddToMarkStack(DependencyNodeCore<DependencyContextType> node, stri
         {
             if (_marker.MarkNode(node, reason1, reason2, reason))
             {
-                PerfEventSource.Log.AddedNodeToMarkStack();
+                if (PerfEventSource.Log.IsEnabled())
+                    PerfEventSource.Log.AddedNodeToMarkStack();
```

Comment:
I think it would be better to just make a new method in PerfEventSource that triggers the AddedNodeToStack event. That way, all the events themselves can be private, and we keep all the logic in one place (PerfEventSource.cs).

Reply:
I can do that, but I would just like to point out that this is an existing common pattern in managed code in this repo. We don't normally wrap these. E.g. here's logging in Task:

https://github.com/dotnet/coreclr/blob/3bdc9f625480f497adfcd2bbfae039fdfd1e73f3/src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs#L1164-L1165

Reply:
Oh I didn't know that. That's fine then.